Source/WebKit/Shared/AuxiliaryProcessExtensions/NetworkingProcessExtension.swift

```diff
@@ -26,13 +26,47 @@
 import ServiceExtensions
 @_spi(Private) import ServiceExtensions
 
+@objc
+@_spi(Private)
+open class Grant: WKGrant {
+    let inner : _Capabilities.Grant
+
+    init(inner: _Capabilities.Grant) {
+        self.inner = inner
+    }
+
+    deinit {
+        do {
+            try invalidate()
+        } catch {
+            NSLog("Failed to invalidate grant")
+        }
+    }
+
+    open func invalidate() throws {
+        try self.inner.invalidate()
+    }
+}
+
 @main
-class NetworkingProcessExtension {
-    required init() {}
+class NetworkingProcessExtension : WKProcessExtension {
+    override required init() {
+        super.init()
+        setSharedInstance(self)
+    }
 }
 
-extension NetworkingProcessExtension: NetworkingServiceExtension {
+extension NetworkingProcessExtension : NetworkingServiceExtension {
     func handle(xpcConnection: xpc_connection_t) {
         handleNewConnection(xpcConnection)
     }
+
+    override func grant(_ domain: String, name: String) -> Any {
+        do {
+            let grant = try self._request(capabilities: _Capabilities.assertion(domain, name))
+            return Grant(inner: grant)
+        } catch {
+            return WKGrant()
+        }
+    }
```

Comment:
That is a good point, although I think it also would be good to avoid the respondsToSelector call. I went ahead and landed as-is, but please let me know if you still think this should be changed, and I can follow up in a new patch :)